From 6519858f618230f2fd00d6aa226ccc12c05c4a33 Mon Sep 17 00:00:00 2001
From: Jakob Draeger <jakob.draeger@web.de>
Date: Tue, 9 May 2017 23:06:04 +0200
Subject: [PATCH] " Added more test cases and added the functionality to rotate
 an image by 180 degrees in org.jis.generator.Generator#rotateImage."

---
 .../src/main/java/org/jis/generator/Generator.java |  9 ++++-
 .../src/test/java/jmjrst/main/RotateImageTest.java | 40 ++++++++++++++++++++++
 2 files changed, 48 insertions(+), 1 deletion(-)

diff --git a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
index ba1bf82..58f9bce 100755
--- a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
+++ b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
@@ -67,6 +67,7 @@ import org.jis.options.Options;
  */
 public class Generator {
   public final static double ROTATE_90  = Math.toRadians(90);
+  public final static double ROTATE_180 = Math.toRadians(180);
   public final static double ROTATE_270 = Math.toRadians(270);
 
   private Main               m;
@@ -723,7 +724,7 @@ public class Generator {
 
     AffineTransform transform = new AffineTransform();
 
-    // get width and height of the origianl image
+    // get width and height of the original image
     int width = image.getWidth(null);
     int height = image.getHeight(null);
 
@@ -734,6 +735,12 @@ public class Generator {
       width = image.getHeight(); // swap
       height = image.getWidth();
     }
+    else if (rotate == Generator.ROTATE_180) {
+      transform.translate(height, 0);
+      transform.rotate(Generator.ROTATE_90);
+      transform.translate(height, 0);
+      transform.rotate(Generator.ROTATE_90);
+    }
     else if (rotate == Generator.ROTATE_270)
     {
       transform.translate(0, width);
diff --git a/iMage/jmjrst.main/src/test/java/jmjrst/main/RotateImageTest.java b/iMage/jmjrst.main/src/test/java/jmjrst/main/RotateImageTest.java
index 20b996a..51f6199 100644
--- a/iMage/jmjrst.main/src/test/java/jmjrst/main/RotateImageTest.java
+++ b/iMage/jmjrst.main/src/test/java/jmjrst/main/RotateImageTest.java
@@ -57,6 +57,46 @@ public class RotateImageTest {
     BufferedImage nullImage = this.generator.rotateImage(null, 0.0);
     assertNull(nullImage);
   }
+  
+  @Test
+  public void rotateImageTest3() {
+    this.setUp();
+    try {
+      BufferedImage illegalImage = this.generator.rotateImage(this.image, 0.7);
+      assertTrue(false);
+      illegalImage.getWidth();
+    } catch (IllegalArgumentException ie) {
+      assertTrue(true);
+      
+    }
+  }
+  
+  @Test
+  public void rotateImageTest4() {
+    this.setUp();
+    int height = this.image.getHeight();
+    BufferedImage rotated90Image = this.generator.rotateImage(this.image, Math.toRadians(90));
+    int width = rotated90Image.getWidth();
+    assertEquals(height, width);
+  }
+  
+  @Test
+  public void rotateImageTest5() {
+    this.setUp();
+    int height = this.image.getHeight();
+    BufferedImage rotated270Image = this.generator.rotateImage(this.image, Math.toRadians(270));
+    int width = rotated270Image.getWidth();
+    assertEquals(height, width);
+  }
+  
+  @Test
+  public void rotateImageTest6() {
+    this.setUp();
+    int height = this.image.getHeight();
+    BufferedImage rotated180Image = this.generator.rotateImage(this.image, Math.toRadians(180));
+    int afterRotationHeight = rotated180Image.getHeight();
+    assertEquals(height, afterRotationHeight);
+  }
 
 
 
-- 
2.10.1

